File: lib/domain/field/entities/field_entity.rb
# frozen_string_literal: true

module Domain
  module Field
    module Entities
      class FieldEntity
        attr_reader :id, :farm_id, :user_id, :name, :description,
                    :created_at, :updated_at, :area, :daily_fixed_cost, :region

        def initialize(id:, farm_id:, user_id:, name:, description:,
                      created_at:, updated_at:, area:, daily_fixed_cost:, region:)
          @id = id
          @farm_id = farm_id
          @user_id = user_id
          @name = name
          @description = description
          @created_at = created_at
          @updated_at = updated_at
          @area = area
          @daily_fixed_cost = daily_fixed_cost
          @region = region
        end

        def display_name
          name.presence || "Field #{id}"
        end
      end
    end
  end
end

File: crates/agrr-domain/src/field/entities/field_entity.rs
/// Ruby: `Domain::Field::Entities::FieldEntity`
#[derive(Debug, Clone, PartialEq)]
pub struct FieldEntity {
    pub id: i64,
    pub farm_id: i64,
    pub user_id: Option<i64>,
    pub name: String,
    pub description: Option<String>,
    pub created_at: Option<String>,
    pub updated_at: Option<String>,
    pub area: Option<f64>,
    pub daily_fixed_cost: Option<f64>,
    pub region: Option<String>,
}

impl FieldEntity {
    pub fn display_name(&self) -> String {
        if self.name.trim().is_empty() {
            format!("Field {}", self.id)
        } else {
            self.name.clone()
        }
    }
}
